src/main/java/mate.academy/lib/Injector.java

```diff
@@ -20,39 +20,38 @@ public static Injector getInjector() {
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        Object classImplementationInstance = null;
+        Object clazzImplementationInstance = null;
         Class<?> clazz = findImplementation(interfaceClazz);
-        Field[] declaredFields = interfaceClazz.getDeclaredFields();
+        Field[] declaredFields = clazz.getDeclaredFields();
         for (Field field : declaredFields) {
             if (field.isAnnotationPresent(Inject.class)) {
-                Object fieldInstance = getInstance(field.getClass());
-                classImplementationInstance = createNewInstance(clazz);
-                field.setAccessible(true);
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
                 try {
-                    field.set(classImplementationInstance, fieldInstance);
+                    field.setAccessible(true);
+                    field.set(clazzImplementationInstance, fieldInstance);
                 } catch (IllegalAccessException e) {
-                    throw new RuntimeException("Can't initialize field value. Class "
-                            + clazz.getName() + " . Field: " + field.getName());
+                    throw new RuntimeException("Can`t initialize field value. Class: "
+                            + clazz.getName() + "Field: " + field.getName());
                 }
             }
         }
-        if (classImplementationInstance == null) {
-            classImplementationInstance = createNewInstance(clazz);
+        if (clazzImplementationInstance == null) {
+            clazzImplementationInstance = createNewInstance(clazz);
         }
-        return classImplementationInstance;
+        return clazzImplementationInstance;
     }
 
     private Object createNewInstance(Class<?> clazz) {
         if (!clazz.isAnnotationPresent(Component.class)) {
             throw new RuntimeException("Cannot create instance of " + clazz.getName()
                     + ",there is no @Component annotation");
         }
-        if (instances.containsKey(clazz)) {
-            return instances.get(clazz);
-        }
-        Constructor<?> constructor = null;
         try {
-            constructor = clazz.getConstructor();
+            if (instances.containsKey(clazz)) {
+                return instances.get(clazz);
+            }
```

Comment:
there is no need to put this `if` into `try` block

the comments on Github can be left to the last line, but Github displays the previous lines as well to show the context